packages/visualizations/src/components/MapPoi/utils.ts

```diff
@@ -1,31 +1,60 @@
-import type { BBox } from 'geojson';
-import type { PoiMapLayer } from './types';
-import type { Color } from '../types';
-import { DEFAULT_COLORS } from './constants';
+import type {
+    CircleLayerSpecification,
+    DataDrivenPropertyValueSpecification,
+    MapOptions,
+    StyleSpecification,
+} from 'maplibre-gl';
 
-export const computeBaseRoundMarkerLayer = (
-    colors: Color[],
-    matchValues: string[],
-    matchKey: string,
-    noMatchColor?: Color
-): PoiMapLayer => {
-    const matchExpression = ['match', ['get', matchKey]];
-    matchValues.forEach((key, i) => {
-        matchExpression.push(key);
-        matchExpression.push(colors[i]);
+import type { Layer, PoiMapData, PoiMapOptions } from './types';
+import { DEFAULT_BASEMAP_STYLE, DEFAULT_ASPECT_RATIO, DEFAULT_BBOX } from './constants';
+
+export const getMapStyle = (style: PoiMapOptions['style']): MapOptions['style'] => {
+    if (!style) return DEFAULT_BASEMAP_STYLE;
+    if (typeof style === 'string') return style;
+    return { ...DEFAULT_BASEMAP_STYLE, ...style };
+};
+export const getMapSources = (sources: PoiMapData['sources']): StyleSpecification['sources'] => {
+    if (!sources) return DEFAULT_BASEMAP_STYLE.sources;
+    return sources;
+};
+
+// Only circle layers are supported
+export const getMapLayers = (layers?: Layer[]): CircleLayerSpecification[] => {
+    if (!layers) return [];
+
+    return layers.map((layer) => {
+        let cirleColor: unknown = layer.color;
+
+        if (layer.colorMatch) {
+            const { key, colors } = layer.colorMatch;
+            const groupByColors = ['match', ['get', key]];
+            Object.keys(colors).forEach((color) => {
+                groupByColors.push(color, colors[color]);
+            });
+            groupByColors.push(layer.color);
+            cirleColor = groupByColors;
+        }
+        const { id, type, source, sourceLayer } = layer;
+        return {
+            id,
+            type,
+            source,
+            ...(sourceLayer ? { 'source-layer': sourceLayer } : undefined),
+            paint: {
+                'circle-radius': 5,
+                // TODO: Better typing
+                'circle-color': cirleColor as DataDrivenPropertyValueSpecification<string>,
+            },
+            filter: ['==', ['geometry-type'], 'Point'],
+        };
     });
-    matchExpression.push(noMatchColor || DEFAULT_COLORS.Blue); // Default fallback color
+};
```

Comment:
What do you think of this typing for the `circleColor` variable? (There's a small typo btw, ~cirleColor~ 👉 circleColor)

``` TS
// ...

import type { Color } from '../types';

// ...

// Only circle layers are supported
export const getMapLayers = (layers?: Layer[]): CircleLayerSpecification[] => {
    if (!layers) return [];

    return layers.map((layer) => {
        let circleColor: DataDrivenPropertyValueSpecification<Color> | Color = layer.color;
        if (layer.colorMatch) {
            const { key, colors } = layer.colorMatch;
            const groupByColors = ['match', ['get', key]];
            Object.keys(colors).forEach((color) => {
                groupByColors.push(color, colors[color]);
            });
            groupByColors.push(layer.color);
            circleColor = groupByColors;
        }
        const { id, type, source, sourceLayer } = layer;
        return {
            id,
            type,
            source,
            ...(sourceLayer ? { 'source-layer': sourceLayer } : undefined),
            paint: {
                'circle-radius': 5,
                'circle-color': circleColor,
            },
            filter: ['==', ['geometry-type'], 'Point'],
        };
    });
};
```

Reply:
Thanks, that's much better 💪